### DISCUSSION TITLE
bus monitor example

### DISCUSSION THREAD

**Discussion by RealByron** [XKNX/xknx#873] [Q&A] [answered]
Hi Everyone, 
I'm new with xknx and would like to make a busmonitor for all bus frame P2P, propertyValueWrite, NetworkParameters R/W, etc...
I'm only get GroupValueWrite and IndividualAddressSerialRead/Response.
Could someone help me ?

**Comment by farmio** [comment, marked as answer]
Hi 👋!
Without knowing what exactly you are trying to do - and how, this excerpt form `3.8.4 Tunnelling §2.2.2 Tunnelling on KNX Data Link Layer` may get you started:

> The KNXnet/IP Tunnelling Server shall only pass those KNX point-to-point addressed telegrams to the KNXnet/IP Tunnelling Client that contain the KNX Individual Address of the KNXnet/IP Tunnelling connection with this KNXnet/IP Tunnelling Client. All KNX telegrams on KNX point-to-multipoint communication (i.e. group addressing) shall be forwarded to the KNXnet/IP Tunnelling Client.

**Reply by RealByron** [reply]
... My bad you're fully rigth. I was trying to spy my bus in busmonitor mode through a tunnelling client.. And it cannot be done like that.
It's time to find my good old usb knx device.
Thank you farmio

**Reply by farmio** [reply]
hm... you could try to use Busmonitor mode if your interface supports it.
https://github.com/XKNX/xknx/blob/3252ef2622c8dfe6ccc974ae1f14e31f74d137cc/xknx/knxip/connect_request.py#L38-L39
I'm not sure what xknx makes out of L_Busmon.ind frames though.
